drivers/ethernet/eth_numaker.c

```diff
@@ -21,22 +21,38 @@
 
 #ifdef CONFIG_SOC_M467
 #include <m460_eth.h>
+#else
+#include <numaker_eth.h>
 #endif
 
 LOG_MODULE_REGISTER(eth_numaker, CONFIG_ETHERNET_LOG_LEVEL);
 
 /* Device EMAC Interface port */
 #define NUMAKER_GMAC_INTF  0
-/* 2KB Data Flash at 0xFF800 */
+/* 2KB Data Flash */
+#if defined(CONFIG_SOC_M467)
 #define NUMAKER_DATA_FLASH (0xFF800U)
+#elif defined(CONFIG_SOC_M55M1XXX)
+#define NUMAKER_DATA_FLASH (0x2FF800U)
+#else
+#error "SoCs without defined data flash for EMAC!"
+#endif
```

Comment:
this seems like something, that could be fetched from the device tree.

Reply:
Yes, I changed to fetch it from device tree at [49d227d](https://github.com/zephyrproject-rtos/zephyr/commit/49d227dd2d5eeddad33cef6b519d4d4450fdb5b6).